scripts/fdbshow

```diff
@@ -0,0 +1,192 @@
+#!/usr/bin/python
+"""
+    Script to show MAC/FDB entries learnt in Hardware
+    
+    usage: fdbshow [-p PORT] [-v VLAN]
+    optional arguments:
+      -p,  --port              FDB learned on specific port: Ethernet0
+      -v,  --vlan              FDB learned on specific Vlan: 1000
+  
+    Example of the output:
+    admin@str~$ fdbshow
+      No.    Vlan  MacAddress         Port
+    -----  ------  -----------------  ----------
+        1    1000  7C:FE:90:80:9F:05  Ethernet20
+        2    1000  7C:FE:90:80:9F:10  Ethernet40
+        3    1000  7C:FE:90:80:9F:01  Ethernet4
+        4    1000  7C:FE:90:80:9F:02  Ethernet8
+    Total number of entries 4
+    admin@str:~$ fdbshow -p Ethernet4
+      No.    Vlan  MacAddress         Port
+    -----  ------  -----------------  ---------
+        1    1000  7C:FE:90:80:9F:01  Ethernet4
+    Total number of entries 1
+    admin@str:~$ fdbshow -v 1001
+    1001 is not in list
+
+"""
+
+import swsssdk
+import argparse
+import json
+import sys
+
+from tabulate import tabulate
+from natsort import natsorted
+
+class FdbShow(object):
+
+    HEADER = ['No.', 'Vlan', 'MacAddress', 'Port']
+    FDB_COUNT = 0
+
+    def __init__(self):
+        super(FdbShow,self).__init__()
+        self.db = swsssdk.SonicV2Connector(host="127.0.0.1")
+        self.get_interface_oid_map()
+        self.get_bridge_port_oid_map()
+        self.fetch_fdb_data()
+        return
+
+    
+    def get_interface_oid_map(self):
+        """
+            Get the Interface names from Counters DB
+        """
+        self.db.connect('COUNTERS_DB')
+        self.if_name_map = self.db.get_all('COUNTERS_DB', 'COUNTERS_PORT_NAME_MAP', blocking=True)
+        self.if_oid_map = {sai_oid: if_name for if_name, sai_oid in self.if_name_map.items()}
+
+
+    def get_bridge_port_oid_map(self):
+        """
+            Get the Bridge port names from ASIC DB
+        """
+        self.db.connect(self.db.ASIC_DB)
+        self.if_br_oid_map = {} 
+        br_port_str = self.db.keys('ASIC_DB', "ASIC_STATE:SAI_OBJECT_TYPE_BRIDGE_PORT:*")
+        if not br_port_str:
+            return
+        
+        offset = len("ASIC_STATE:SAI_OBJECT_TYPE_BRIDGE_PORT:")
+        oid_pfx = len("oid:0x")
+        for s in br_port_str:
+            br_port_id = s[(offset + oid_pfx):]
+            ent = self.db.get_all('ASIC_DB', s, blocking=True)
+            port_id = ent[b"SAI_BRIDGE_PORT_ATTR_PORT_ID"][oid_pfx:]
+            self.if_br_oid_map[br_port_id] = port_id 
+
+
+    def fetch_fdb_data(self):
```

Comment:
>fetch_fdb_data [](start = 8, length = 14)

I notice code duplication with sonic-snmpagent. Could you reuse code?

Reply:
I checked this in detail to see if we can reuse some functionality but there is tight coupling with MIB and SNMP functionality. Since this is a standalone functionality, thought of keeping it separate.

Reply:
Modified to use common library API